buildSrc/src/main/groovy/BuildkitBuild.groovy

```diff
@@ -0,0 +1,357 @@
+                imageRepo.add([name: targetRepo, tag: targetTags])
+            } else {
+                imageRepo.add([name: targetRepo, tag: ""])
+            }
+        } else if (dockerHubPublish.get()) {
+            targetRepo = "corda"
+            targetTags = ["${version}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else if (preTest.get()) {
+            targetRepo = "corda-os-docker-pre-test.software.r3.com"
+            targetTags = ["preTest-${tagPrefix}${version}", "preTest-${tagPrefix}${gitRevision}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else if (releaseType == 'RC' || releaseType == 'GA') {
+            targetRepo = "corda-os-docker-stable.software.r3.com"
+            targetTags = ["${tagPrefix}latest", "${tagPrefix}${version}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else if (releaseType == 'BETA' && !nightlyBuild.get()) {
+            targetRepo = "corda-os-docker-unstable.software.r3.com"
+            targetTags = ["${tagPrefix}unstable", "${tagPrefix}${gitRevision}", "${version}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else if (releaseType == 'ALPHA' && !nightlyBuild.get()) {
+            targetRepo = "corda-os-docker-dev.software.r3.com"
+            targetTags = ["${tagPrefix}${gitRevision}", "${version}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else if (releaseType == 'BETA' && nightlyBuild.get()) {
+            targetRepo = "corda-os-docker-nightly.software.r3.com"
+            targetTags = ["${tagPrefix}nightly", "${tagPrefix}nightly-${timeStamp}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else if (releaseType == 'ALPHA' && nightlyBuild.get()) {
+            targetRepo = "corda-os-docker-nightly.software.r3.com"
+            targetTags = ["${tagPrefix}nightly-${version}", "${tagPrefix}nightly-${gitRevision}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        } else {
+            targetRepo = "corda-os-docker-dev.software.r3.com"
+            targetTags = ["latest-local", "${version}", "${gitRevision}"]
+            imageRepo.add([name: targetRepo, tag: targetTags])
+        }
+
+        List<String> javaArgs = new ArrayList<String>(arguments.get())
+        javaArgs.add("-Dlog4j2.debug=\${ENABLE_LOG4J2_DEBUG:-false}")
+
+        def baseImageName = "${baseImageTag.get().empty ? baseImageName.get() : "${baseImageName.get()}:${baseImageTag.get()}"}"
+        def entryName = overrideEntryName.get().empty ? projectName : overrideEntryName.get()
+
+        dockerAuth.add(targetRepo)
+
+        for (repo in dockerAuth) {
+            String[] cmd = ['docker', 'login', "${repo}", "-u ${registryUsername.get()}", "-p ${registryPassword.get()}"]
+            ExecShellCommand(cmd)
+        }
+
+        String[] baseCommand = []
+        String[] opts = []
+        String[] commandTail = []
```

Comment:
```suggestion
        String[] baseCommand
        String[] opts
        String[] commandTail
```
... also IntelliJ reports a number of warning like this in this file - do you want to address them?